docs/design/scheduler_object_states.md

```diff
@@ -50,14 +50,15 @@ This state is part of the normal scheduling cycle.
 * Running: The state in which the application will spend most of its time.
 Containers/pods can be added to and removed from the application. 
 This state is part of the normal scheduling cycle.
-* Waiting: An application that has no pending requests or running containers/pod will be waiting.
+* Completing: An application that has no pending requests or running containers/pod will be completing.
 This state shows that the application has not been marked completed yet but currently is not actively being scheduled.
-* Completed: An application is considered completed when it has been in the waiting state for a defined time period.
-From this state the application can only move to the Expired state and it cannot move back into any of scheduling states (Running or Waiting)
+* Completed: An application is considered completed when it has been in the completing state for a defined time period.
+From this state the application can only move to the Expired state, and it cannot move back into any of scheduling states (Running or Completing)
 The current timeout is set to 30 seconds.
 * Expired: The completed application is tracked for a period of time, after that is expired and deleted from the scheduler.
 This is a final state and after this state the application cannot be tracked anymore. 
-* Killed: Removed by the resource manager at the request of an administrator or the user running the application.
+* Failing: An application marked for failing, what still has some allocations or asks what needs to be cleaned up before entering into the Failed state.
```

Comment:
can we add a bit more message about "when the app can be marked as Failing"?
Current context only explains what happens if the app is marked as Failing but not explained why.